spec/outputs/graphite_spec.rb

```diff
@@ -89,122 +43,66 @@
 
-      context "no match" do
-
-        let(:config) do  <<-CONFIG
-          input {
-            generator {
-              message => "foo=123 bar=42"
-              count => 1
-              type => "generator"
-            }
-          }
-
-          filter {
-            kv { }
-          }
-
-          output {
-            graphite {
-              host => "localhost"
-              port => #{port}
-              fields_are_metrics => true
-              include_metrics => ["notmatchinganything"]
-              metrics_format => "foo.bar.sys.data.*"
-              debug => true
-            }
-          }
-        CONFIG
-        end
+      it "match the generated bar key" do
+        expect(lines['foo.bar.sys.data.bar']).to match(/^foo.bar.sys.data.bar 42.0 \d{10,}\n$/)
+      end
+    end
 
-        it "generate no event" do
-          expect(server.empty?).to eq(true)
-        end
+    context "no match" do
+
+      let(:subject) { LogStash::Outputs::Graphite.new("host" => "localhost",
+                                                      "port" => port,
+                                                      "fields_are_metrics" => true,
+                                                      "include_metrics" => ["notmatchinganything"],
+                                                      "metrics_format" => "foo.bar.sys.data.*") }
+
+      let(:event) { LogStash::Event.new("foo" => "123", "bar" => "42") }
+
+      it "generate no event" do
+        expect(server.empty?).to eq(true)
       end
+    end
 
-      context "match a key with invalid metric_format" do
-
-        let(:config) do <<-CONFIG
-          input {
-            generator {
-              message => "foo=123"
-              count => 1
-              type => "generator"
-            }
-          }
-
-          filter {
-            kv { }
-          }
-
-          output {
-            graphite {
-                host => "localhost"
-                port => #{port}
-                fields_are_metrics => true
-                include_metrics => ["foo"]
-                metrics_format => "invalidformat"
-                debug => true
-            }
-          }
-        CONFIG
-        end
+    context "match a key with invalid metric_format" do
 
-        it "match the foo key" do
-          lines = server.pop
-          expect(lines).to match(/^foo 123.0 \d{10,}\n$/)
-        end
+      let(:subject) { LogStash::Outputs::Graphite.new("host" => "localhost",
+                                                      "port" => port,
+                                                      "fields_are_metrics" => true,
+                                                      "include_metrics" => ["foo"],
+                                                      "metrics_format" => "invalidformat") }
+
+      let(:event) { LogStash::Event.new("foo" => "123") }
+
+      it "match the foo key" do
+        lines = server.pop
+        expect(lines).to match(/^foo 123.0 \d{10,}\n$/)
```

Comment:
some lines before you removed the lines variables, but not here. I like to have lines as in here, but even if we do it or not, we should be consistent :-)